From ead4a8e52894b906acbc17c51a755fdd628fa04a Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 12 Sep 2023 17:07:46 +0200
Subject: [PATCH 408/954] feat(stm32mp2): introduce STM32MP_M33_TDCID flag

If Cortex-A35 does not own the Trusted Domain Compartment ID (TDCID), it
means the Cortex-M33 has it, and several secure resources can no more be
accessed from Cortex-A35. The STM32MP_M33_TDCID will be used to stub
the resources that are no more accessible.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 66df81fec85633f1c5f4f4f3de16b19334b93b95 feat(stm32mp2): introduce STM32MP_M33_TDCID flag]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Id3b61a04ac7851030eb1362eae2d00fad01f2221
---
 docs/plat/st/stm32mp2.rst    | 2 ++
 plat/st/stm32mp2/platform.mk | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/docs/plat/st/stm32mp2.rst b/docs/plat/st/stm32mp2.rst
index e914832c1b..f2c0846a69 100644
--- a/docs/plat/st/stm32mp2.rst
+++ b/docs/plat/st/stm32mp2.rst
@@ -66,6 +66,8 @@ Dedicated STM32MP2 build flags:
 
 - | ``STM32MP_DDR_FIP_IO_STORAGE``: to store DDR firmware in FIP.
   | Default: 1
+- | ``STM32MP_M33_TDCID``: Enable this flag if Cortex-A35 does not have the Trusted Domain Compartment ID (owned by Cortex-M33)
+  | Default: 0
 - | ``STM32MP25``: to select STM32MP25 variant configuration.
   | Default: 1
 
diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index a1a727a68a..c866d77ad8 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -27,6 +27,7 @@ PSCI_OS_INIT_MODE		:= 1
 DTB_FILE_NAME			?=	stm32mp257f-ev1.dtb
 
 STM32MP25			:=	1
+STM32MP_M33_TDCID		?=	0
 
 STM32MP_USE_EXTERNAL_HEAP :=	1
 
@@ -121,6 +122,7 @@ $(eval $(call assert_booleans,\
 		STM32MP_DDR3_TYPE \
 		STM32MP_DDR4_TYPE \
 		STM32MP_LPDDR4_TYPE \
+		STM32MP_M33_TDCID \
 		STM32MP_USE_EXTERNAL_HEAP \
 		STM32MP25 \
 )))
@@ -150,6 +152,7 @@ $(eval $(call add_defines,\
 		STM32MP_DDR3_TYPE \
 		STM32MP_DDR4_TYPE \
 		STM32MP_LPDDR4_TYPE \
+		STM32MP_M33_TDCID \
 		STM32MP_USE_EXTERNAL_HEAP \
 		STM32MP25 \
 )))
-- 
2.53.0

